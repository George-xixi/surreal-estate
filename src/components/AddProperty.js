import React, { useState } from "react";
import addProperty from "../requests/addProperty";
import "../styles/add-property.css";
import Alert from "./Alert";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "Flat",
      bedrooms: 0,
      bathrooms: 0,
      price: 0,
      email: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);
  const handleAddProperty = (e) => {
    e.preventDefault();
    setAlert({ message: "", isSuccess: false });
    addProperty(fields, setAlert);
  };
  const handleFieldChange = (e) => {
    setFields({
      ...fields,
      [e.target.name]: e.target.value,
    });
  };

  return (
    <div className="add-property">
      <h2 className="add-property__heading">Add Property</h2>
      <Alert message={alert.message} success={alert.isSuccess} />
      <form onSubmit={handleAddProperty} className="add-property__form">
        <div className="form__title">
          <label htmlFor="title" data-testId="first-label">
            <p>Give your listing a title</p>
            <input
              id="title"
              name="title"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form__selector selector__city">
          <label htmlFor="city">
            <p>City</p>
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
        </div>
        <div className="form__selector selector__type">
          <label htmlFor="type">
            <p>Type</p>
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
        </div>
        <div className="form__number-input number__beds">
          <label htmlFor="bedrooms">
            <p>Beds</p>
            <input
              id="bedrooms"
              name="bedrooms"
              type="number"
              inputMode="numeric"
              min={0}
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form__number-input number__baths">
          <label htmlFor="bathrooms">
            <p>Baths</p>
            <input
              id="bathrooms"
              name="bathrooms"
              type="number"
              inputMode="numeric"
              min={0}
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form__number-input number__price">
          <label htmlFor="price">
            <p>Price</p>
            <input
              id="price"
              name="price"
              type="number"
              inputMode="numeric"
              min={0}
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form__email-input">
          <label htmlFor="email">
            <p>Email</p>
            <input
              id="email"
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
        </div>
        <div className="form__button">
          <button type="submit">Add</button>
        </div>
      </form>
    </div>
  );
};

export default AddProperty;
