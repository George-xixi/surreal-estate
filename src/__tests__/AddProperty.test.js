import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  it("renders AddProperty page correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const headingElement = screen.getByText("Add Property");
    const formFirstQuestion = screen.getByText("Give your listing a title");

    expect(asFragment()).toMatchSnapshot();
    expect(headingElement).toBeInTheDocument();
    expect(formFirstQuestion).toHaveAttribute("for", "title");
  });
});
