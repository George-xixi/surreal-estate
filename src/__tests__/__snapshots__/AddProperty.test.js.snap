// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders AddProperty page correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <h2
      class="add-property__heading"
    >
      Add Property
    </h2>
    <form
      class="add-property__form"
    >
      <div
        class="form__title"
      >
        <label
          for="title"
        >
          Give your listing a title
        </label>
        <input
          id="title"
          name="title"
          value=""
        />
      </div>
      <div
        class="form__selector selector__city"
      >
        <label
          for="city"
        >
          City
        </label>
        <select
          id="city"
          name="city"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </div>
      <div
        class="form__selector selector__type"
      >
        <label
          for="type"
        >
          Type
        </label>
        <select
          id="type"
          name="type"
        >
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </div>
      <div
        class="form__number-input number__beds"
      >
        <label
          for="bedrooms"
        >
          Beds
        </label>
        <input
          id="bedrooms"
          inputmode="numeric"
          min="0"
          name="bedrooms"
          type="number"
          value="0"
        />
      </div>
      <div
        class="form__number-input number__baths"
      >
        <label
          for="bathrooms"
        >
          Baths
        </label>
        <input
          id="bathrooms"
          inputmode="numeric"
          min="0"
          name="bathrooms"
          type="number"
          value="0"
        />
      </div>
      <div
        class="form__number-input number__price"
      >
        <label
          for="price"
        >
          Price
        </label>
        <input
          id="price"
          inputmode="numeric"
          min="0"
          name="price"
          type="number"
          value="0"
        />
      </div>
      <div
        class="form__email-input"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          value=""
        />
      </div>
      <div
        class="form__button"
      >
        <button
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Navbar renders Navbar correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <svg
      aria-label="Home icon"
      class="home-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m12.223 11.641-.223.22-.224-.22a2.224 2.224 0 0 0-3.125 0 2.13 2.13 0 0 0 0 3.07L12 18l3.349-3.289a2.13 2.13 0 0 0 0-3.07 2.225 2.225 0 0 0-3.126 0z"
      />
      <path
        d="m21.707 11.293-9-9a.999.999 0 0 0-1.414 0l-9 9A1 1 0 0 0 3 13h1v7c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-7h1a1 1 0 0 0 .707-1.707zM18.001 20H6v-9.585l6-6 6 6V15l.001 5z"
      />
    </svg>
    <h1>
      surreal estate
    </h1>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          view properties
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-property"
        >
          add property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
